pkg/infra/tracing/tracing.go

```diff
@@ -52,6 +53,21 @@ func ProvideService(cfg *setting.Cfg) (Tracer, error) {
 	return ots, ots.initOpentelemetryTracer()
 }
 
+type traceKey struct{}
+type traceValue struct {
+	ID        string
+	IsSampled bool
+}
+
+func TraceIDFromContext(c context.Context, sampled bool) string {
```

Comment:
```suggestion
func TraceIDFromContext(c context.Context, requireSampled bool) string {
```

I think `sampled` is a bit ambiguous here, not sure what to call it instead to make it clear that this is a filtering argument and not whether or not the trace is sampled. WDYT?